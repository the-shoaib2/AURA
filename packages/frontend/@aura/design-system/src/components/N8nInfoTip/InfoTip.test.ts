import { render } from '@testing-library/vue';

import auraInfoTip from './InfoTip.vue';

const slots = {
	default: ['Need help doing something?', '<a href="/docs" target="_blank">Open docs</a>'],
};
const stubs = ['aura-tooltip', 'aura-icon'];

describe('auraInfoTip', () => {
	it('should render correctly as note', () => {
		const wrapper = render(auraInfoTip, {
			slots,
			global: {
				stubs,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should render correctly as tooltip', () => {
		const wrapper = render(auraInfoTip, {
			slots,
			props: {
				type: 'tooltip',
			},
			global: {
				stubs,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should render correctly with a specific size', () => {
		const wrapper = render(auraInfoTip, {
			slots,
			props: {
				size: 'large',
			},
			global: {
				stubs,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
