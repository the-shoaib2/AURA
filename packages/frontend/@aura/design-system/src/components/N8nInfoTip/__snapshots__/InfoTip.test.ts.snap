// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`auraInfoTip > should render correctly as note 1`] = `
"<div class="aura-info-tip infoTip info note bold">
  <!-- Note that the branching is required to support displaying
		 the slot either in the tooltip of the icon or following it --><span class="iconText"><aura-icon-stub icon="info" spin="false" color="text-base"></aura-icon-stub><span>Need help doing something?<a href="/docs" target="_blank">Open docs</a></span></span>
</div>"
`;

exports[`auraInfoTip > should render correctly as tooltip 1`] = `
"<div class="aura-info-tip infoTip info tooltip bold">
  <!-- Note that the branching is required to support displaying
		 the slot either in the tooltip of the icon or following it --><span class="iconText el-tooltip__trigger el-tooltip__trigger"><aura-icon-stub icon="info" spin="false" color="text-base"></aura-icon-stub></span>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`auraInfoTip > should render correctly with a specific size 1`] = `
"<div class="aura-info-tip infoTip info note bold">
  <!-- Note that the branching is required to support displaying
		 the slot either in the tooltip of the icon or following it --><span class="iconText"><aura-icon-stub icon="info" size="large" spin="false" color="text-base"></aura-icon-stub><span>Need help doing something?<a href="/docs" target="_blank">Open docs</a></span></span>
</div>"
`;
